src/include/daos_srv/daos_server.h

```diff
@@ -404,14 +408,18 @@ struct dss_acc_task {
  */
 int dss_acc_offload(struct dss_acc_task *at_args);
 
-/** Different type of ES pools, there are 3 pools for now
+/**
+ * Different type of ES pools, there are 4 pools for now
  *
- *  DSS_POOL_PRIV     Private pool: I/O requests will be added to this pool.
- *  DSS_POOL_SHARE    Shared pool: Other requests and ULT created during
- *                    processing rpc.
- *  DSS_POOL_REBUILD  rebuild pool: pools specially for rebuild tasks.
+ *  DSS_POOL_FIRST	The highest priority pool. ULTs in this pool will be
+ *			scheduled firstly.
+ *  DSS_POOL_PRIV	Private pool: I/O requests will be added to this pool.
+ *  DSS_POOL_SHARE	Shared pool: Other requests and ULT created during
+ *			processing rpc.
+ *  DSS_POOL_REBUILD	rebuild pool: pools specially for rebuild tasks.
```

Comment:
I think we probably should use this opportunity to cleanup this a bit, these types are from different dimensions:
- FIRST: defined by priority
- PRIV & SHARE: defined by whether the pool can accept ULTs created by different xstreams
- REBUILD: defined by job type

Should we figure out the dominant (I suppose it's job type) and redefine them, like:
- DSS_ABT_POOL_URGENT: urgent jobs like dtx resync;
- DSS_ABT_POOL_IO: regular I/O
- DSS_ABT_POOL_MISC: misc jobs could be created by different xstreams
- DSS_ABT_POOL_REBUILD: rebuild
- DSS_ABT_POOL_AGG: aggregation
- ...

Reply:
we probably need not separate pool for aggregate, for aggregate there is DSS_ULT_AGGREGATE ULT type.
different ABT pools can with different access policy (MPSC, PRIV) and schedule priority.
agreed that _URGENT is a better name than _FIRST, and looks fine to rename PRIV/SHARE to IO/MISC as well.

So we will have 4 pool types: URGENT, IO (with ABT_POOL_ACCESS_PRIV access), and MISC/REBUILD (with ABT_POOL_ACCESS_MPSC access). uses pool type to select with pool to use. URGENT with highest priority, and REBUILD with lower priority than IO/MISC.
And another dimension of ult_type, together with tgt_id to select which XS to use.

any other suggestions?

Reply:
Yes, I agree with you that we don't have to create ABT pool for aggregation in this patch, but I'm afraid that eventually we need a dedicated ABT pool for aggregation, otherwise, I don't see how can we prioritize aggregation ULT and other type of ULTs within same ABT pool efficiently.

ult_type looks to me is only used for selecting xstream and ABT pool. and all the crucial ULTs (io, rebuild, aggregate, etc.) need their dedicated ABT pools, so than we could easily prioritize/throttle them by our own scheduler.

4 pool types looks good for now, we can expand it later when necessary. Thanks.

Reply:
thx, we probably can have 4 pools per each XS with name of -
URGENT_PRIV(dtx resync), NORMAL_PRIV(IO), NORMAL_SHARE(IO, misc ULT), LOW_SHARE(rebuild, aggregate ULT).
And probably can select both pool type and XS based on ULT type and tgt_idx.

Reply:
I still think that aggregate ULT should use a dedicated ABT pool eventually, and DSS_POOL_PRIV, DSS_POOL_SHARE should be renamed, but we can do such cleanup in later patch.

Reply:
not very sure if aggregate can use same pool as rebuild (can rename if can use one), more pool will add a little overhead to the scheduler.
agreed probably can have a better name for PRIV and SHARE

Reply:
for the pool name PRIV/SHARE, you suggested the name of IO/MISC, a problem is some IO request served by PRIV POOL but some others by SHARE pool. I wanted to change it to be NORMAL_PRIV/NORMAL_SHARE but seems not good enough, so did not change it.